### DISCUSSION TITLE
Battle test ts configuration

### DISCUSSION THREAD

**Discussion by gavinmcfarland** [gavinmcfarland/plugma#20] [Ideas]
Need to check the target and other settings that's needed for Plugin API code, vs code needed for frontend.

If people notice issues, please let me know so I can document them and improve the default configuration.

**Comment by DigitalNaut** [comment]
One thing I've run into is that the type definitions don't exclude browser types that are not part of the sandbox, like `AbortController`.

**Comment by gavinmcfarland** [comment]
Thank you. I'll look into it. I'm presuming this is on the UI side?

**Reply by DigitalNaut** [reply]
The UI side can create `AbortController` without a problem because of the iframe. It's only the sandbox that includes the browser type definitions.

Here's `main.ts` file from a blank Plugma project:

<img width="464" height="212" alt="image" src="https://github.com/user-attachments/assets/92fbceb1-21d6-42f2-86cb-7143d4f673fa" />

<img width="780" height="207" alt="image" src="https://github.com/user-attachments/assets/dcda8919-b1dc-41c9-a0aa-7873a9819ea9" />

I checked the base plugin project created by Figma, and it correctly excludes these types. Here's the `code.ts` file for it:

<img width="595" height="158" alt="image" src="https://github.com/user-attachments/assets/ccc9a39a-2127-4c21-9d97-76ce4a154c25" />

**Reply by gavinmcfarland** [reply]
Ah, I see. Ok cool. I'll look into this. Thank you, much appreciated!
